spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindHidingSubClass.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.Detector;
+import edu.umd.cs.findbugs.SourceLineAnnotation;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import org.apache.bcel.generic.BasicType;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.generic.Type;
+import org.apache.bcel.Const;
+
+import java.util.Arrays;
+
+/**
+ * This detector finds all the methods of a subclass which are hiding the static methods of the superclass and
+ * tries to invoke that method using the instance variable.
+ * Please see @see <a href="https://wiki.sei.cmu.edu/confluence/display/java/MET07-J.+Never+declare+a+class+method+that+hides+a+method+declared+in+a+superclass+or+superinterface">SEI CERT MET07-J</a>
+ * @author Nazir, Muhammad Zafar Iqbal
+ */
+public class FindHidingSubClass implements Detector {
+    private final BugReporter bugReporter;
+
+    public FindHidingSubClass(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    /*
+     * Only the super classes are analyzed, not the super interfaces
+     * Reason: when I looked in detail of the super interfaces,
+     * I found there is no point to analyze as whenever we have method hiding due to super interfaces,
+     * We can call the method by fully qualified name only (as Interface can't be instantiated).
+     * Calling a hidden method by fully qualified name is a compliant test.
+     * See @see <a href="https://wiki.sei.cmu.edu/confluence/display/java/MET07-J.+Never+declare+a+class+method+that+hides+a+method+declared+in+a+superclass+or+superinterface">SEI CERT MET07-J</a>
+     * Thereof, it is useless to analyze super interfaces in the detector.
+     */
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        //First, I get the subclass and superclasses in variables.
+        //This is the current class. Named it subClass to better depict the idea of sub and super class.
+        JavaClass subClass = classContext.getJavaClass();
+        //No need for null check as every class has at least `Object` as its super class.
+        JavaClass[] superClasses = null;
+        try {
+            superClasses = subClass.getSuperClasses();
+        } catch (ClassNotFoundException e) {
+            AnalysisContext.reportMissingClass(e);
```

Comment:
Can't we have a problem here if an exception is thrown and `superClasses` remains null?

Reply:
`superClasses` is a local variable. If we hit an exception, we just report the error by calling another function and do not use the `superClasses` variable anymore so there can be no problem.

Reply:
Unless I misunderstood `AnalysisContext.reportMissingClass(e);` stores the exception and then the execution continues, then line 73 `for (JavaClass superClass : superClasses) {` is executed but `superClasses` remained null

Reply:
I fixed this by returning in case the null an exception occurs and the build is successful and it passes all the tests.